File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarraySum = function(nums, k) {

    let prefixSum = 0;
    let count = 0;
    let map = new Map();
    map.set(0, 1);
    
    for (let i = 0; i < nums.length; i++) {
        prefixSum += nums[i];
        let target = prefixSum - k;
        
        if (map.has(target)) {
            count += map.get(target);
        }
        
        map.set(prefixSum, (map.get(prefixSum) || 0) + 1);
    }
    
    return count;

}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
      
        int prefixSum = 0;
        Map<Integer, Integer> map = new HashMap<>();
        int result = 0;
        map.put(0,1);

        for(int num : nums)
        {
            prefixSum += num;
            int value = prefixSum - k;

            if(map.containsKey(value))
            {
                result += map.get(value);
            }
                
            map.put(prefixSum, map.getOrDefault(prefixSum,0) + 1);
        }

        return result;
    }
}